src/app/clients/ClientsPage.tsx

```diff
@@ -0,0 +1,185 @@
+import { ClientsTableRow } from "@/app/clients/getClientsTableData";
+import FlaggedForAttentionIcon from "@/components/Icons/FlaggedForAttentionIcon";
+import PhoneIcon from "@/components/Icons/PhoneIcon";
+import CongestionChargeAppliesIcon from "@/components/Icons/CongestionChargeAppliesIcon";
+import DeliveryIcon from "@/components/Icons/DeliveryIcon";
+import CollectionIcon from "@/components/Icons/CollectionIcon";
+import ExpandedClientDetails from "@/app/clients/ExpandedClientDetails";
+import ExpandedClientDetailsFallback from "@/app/clients/ExpandedClientDetailsFallback";
+import Icon from "@/components/Icons/Icon";
+import { faUser } from "@fortawesome/free-solid-svg-icons";
+import Modal from "@/components/Modal/Modal";
+import { Schema } from "@/supabase";
+import TableSurface from "@/components/Tables/TableSurface";
+
+const collectionCentreToAbbreviation = (
+    collectionCentre: Schema["parcels"]["collection_centre"]
+): string => {
+    switch (collectionCentre) {
+        case "Brixton Hill - Methodist Church":
+            return "BH-MC";
+        case "Clapham - St Stephens Church":
+            return "CLP-SC";
+        case "N&B - Emmanuel Church":
+            return "NAB-EC";
+        case "Streatham - Immanuel & St Andrew":
+            return "STM-IS";
+        case "Vauxhall Hope Church":
+            return "VHC";
+        case "Waterloo - Oasis":
+            return "WAT-OA";
+        case "Waterloo - St George the Martyr":
+            return "WAT-SG";
+        case "Waterloo - St Johns":
+            return "WAT-SJ";
+        default:
+            return "-";
+    }
+};
+
+export const clientTableHeaderKeysAndLabels: TableHeaders = [
+    ["iconsColumn", ""],
+    ["fullName", "Name"],
+    ["familyCategory", "Family"],
+    ["addressPostcode", "Postcode"],
+    ["deliveryCollection", ""],
+    ["packingDate", "Packing Date"],
+    ["packingTimeLabel", "Time"],
+    ["lastStatus", "Last Status"],
+];
+
+const toggleableHeaders = [
+    "fullName",
+    "familyCategory",
+    "addressPostcode",
+    "packingDate",
+    "packingTimeLabel",
+    "lastStatus",
+];
+
+const clientTableColumnStyleOptions = {
+    iconsColumn: {
+        width: "3em",
+    },
+    fullName: {},
+    familyCategory: {
+        hide: 550,
+    },
+    addressPostcode: {
+        hide: 800,
+    },
+    deliveryCollection: {},
+    packingDate: {
+        hide: 800,
+    },
+    packingTimeLabel: {
+        hide: 800,
+        minWidth: "6rem",
+    },
+    lastStatus: {},
+};
+
+interface Props {
+    clientsTableData: ClientsTableRow[];
+}
+
+const ClientsPage: React.FC<Props> = (props) => {
+    const [selectedParcelId, setSelectedParcelId] = useState<string | null>(null);
+
+    const theme = useTheme();
+
+    const rowToIconsColumn = (row: Row): React.ReactElement => {
+        const data = row.data as ClientsTableRow;
+
+        return (
+            <>
+                {data.flaggedForAttention ? <FlaggedForAttentionIcon /> : <></>}
```

Comment:
I've seen people use `boolean && <thing/>` so empty fragments aren't lying around - what's your thoughts?

Reply:
Personally, I prefer the use of the ternary as I'm not particularly a fan of using `&&` outside of boolean expressions and makes it less explicit (at least for me) as to what is going on